fastparquet/api.py

```diff
@@ -319,6 +319,12 @@ def __len__(self):
         else:
             return 0
 
+
+    def __bool__(self):
+        """Return True if path defined."""
+        return bool(self.fn)
```

Comment:
Can you please point me to where in dask this is important? It isn't obvious what bool(ParquetFile) should do; we certainly don't want to load any data, and I suppose a dataset with no row-groups still contains something, if only metadata. For reference, numpy and pandas simply do not allow bool().

Reply:
Hi @martindurant
Long time I did not contribute! :) I hope you are well.

> Can you please point me to where in dask this is important?

With `__len__` implemented, but not `__bool__` then this test fails in `fastparquet.py`, [line 956](https://github.com/dask/dask/blob/98273f749e0e0c515346ec3e4fd178840f4a2695/dask/dataframe/io/parquet/fastparquet.py#L956).
```python
            elif parquet_file:

                row_groups = []
                for piece in pieces:
                    # `piece[1]` will contain actual row-group objects,
                    # but they may be pickled
                    rgs = piece[0]
                    if isinstance(rgs, bytes):
                        rgs = pickle.loads(rgs)
                    row_groups += rgs
                update_parquet_file = True
```
This results in 23 failing tests, as reported in automatic testing [here](https://github.com/yohplala/fastparquet/runs/7646825830?check_suite_focus=true), before I added `__bool__`.

Reply:
Probably should have been `parquet_file is not None` or something like that. But your fix is fine and makes sense. It could even be simply `return True`. Have you any idea why it evaluated True before `__len__`?

Reply:
> Have you any idea why it evaluated True before `__len__`?

No, but because of your questions, I googled the topic, and here is what I can find, that indeed answers your question:

> Here are a few cases, in which Python’s bool() method returns false. **Except these all other values return True.** 
>
> If a False value is passed.
> If None is passed.
> If an empty sequence is passed, such as (), [], ”, etc
> If Zero is passed in any numeric type, such as 0, 0.0 etc
> If an empty mapping is passed, such as {}.
> If Objects of Classes having __bool__() or __len()__ method, returning 0 or False

Indeed, none of the few case seems to correspond, so by default `bool()` would return `True`.

Reply:
> It could even be simply `return True`

You are right.
I am not on my coding PC, but later today I will be. I will make the correction.
Thanks,

Reply:
> It could even be simply `return True`

Ok, done.
The failing testing appears related directly to a problem with the testing environment / network, not this development.
![Screenshot from 2022-08-03 20-41-16](https://user-images.githubusercontent.com/13206058/182685012-5b373700-457f-43e8-9f03-abd4648a9097.png)